scripts/upgrade

```diff
@@ -101,25 +98,63 @@ then
   if 	[ -d "$backup_squelettes" ]
   then
     cp -ar $backup_squelettes $final_path/www/squelettes
-    ynh_secure_remove  $backup_squelettes
-  fi
+   fi
 
   if [ -e "$backup_bdd" ]
   then
 	  cp -a $backup_bdd $final_path/association.sqlite
-    ynh_secure_remove $backup_bdd
   fi
 
   chown -R $app:$app $final_path
+  chmod 755 $final_path
 
-  key=$(ynh_string_random --length=50)
-  cp -a $final_path/config.dist.php $final_path/config.local.php
-  ynh_replace_string --match_string="const SECRET_KEY = '3xUhIgGwuovRKOjVsVPQ5yUMfXUSIOX2GKzcebsz5OINrYC50r';" --replace_string="const SECRET_KEY = '$key';" --target_file="$final_path/config.local.php"
-  if [ "$path_url" == "/" ]; then
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url';" --target_file="$final_path/config.local.php"
-  else
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url/';" --target_file="$final_path/config.local.php"
+  # Create the visitors permission if needed
+   visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
+  if [[ $visitors_enabled -eq 0 ]]; then
+    ynh_permission_update --permission "main" --add "visitors"
   fi
+
+  ynh_systemd_action --service_name=nginx --action=reload
+  # Finalize the upgrade
+  ynh_local_curl "/index.php"
+  sleep 5
+  ynh_local_curl "/index.php"
+  upgrade_type="UPGRADE_APP"
+
+  if [[ $visitors_enabled -eq 0 ]]; then
+    ynh_permission_update --permission "main" --remove "visitors"
+  fi
+fi
+#=================================================
+# UPGRADE TO 1.1.0 if necessary
+#=================================================
+
+if ynh_compare_current_package_version --comparison lt --version 1.1.0~ynh1
+then
+  ynh_script_progression --message="Upgrading upgrade to 1.1.0" --weight=5
+  # Download, check integrity, uncompress and patch the source from 0.9.8.src
+  ynh_setup_source --keep="config.local.php" --dest_dir="$final_path" --source_id="1.1.0"
+  # Set permissions on app files
+  ynh_system_user_create --username=$app
+  ynh_add_nginx_config
+  ynh_add_fpm_config --package="$extra_php_dependencies"
+  phpversion=$(ynh_app_setting_get --app=$app --key=phpversion)
+  ynh_script_progression --message="restore data..." --weight=10
+
+  backup_bdd=/tmp/association.sqlite
+  backup_squelettes=/tmp/squelettes
+
+  if  [ -d "$backup_squelettes" ]
+  then
+    cp -ar $backup_squelettes $final_path/www/squelettes
+  fi
+
+  if [ -e "$backup_bdd" ]
+  then
+    cp -a $backup_bdd $final_path/association.sqlite
+  fi
+
+  chown -R $app:$app $final_path
   chmod 755 $final_path
 
   # Create the visitors permission if needed
```

Comment:
(same here as before, but i can't comment the line below ... question regarding temporarily enabling visitors)

Reply:
here, it's just a temporally permissions, we change the permissions on files just with the last upgrade to 1.1.* and the owner too, before the owner was app:app, now www-data:app....

Reply:
You think we can comment this line ? I think is important to keep these lines for the command `ynh_local_curl "/index.php"` which finalize the upgrade...

Reply:
Here you are talking about the same conditions for pre-upgrade 1.1.0 ? if it is I change also with the same code.

Reply:
I think it is ! so I resolve this also

Reply:
resolved with https://github.com/YunoHost-Apps/garradin_ynh/pull/63/commits/0e6e4467cf7525e18b3d5cc558a01d3725216a02